staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1138,21 +1143,34 @@ func (c *cacheWatcher) convertToWatchEvent(event *watchCacheEvent) *watch.Event
 		return nil
 	}
 
+	var watchEvent watch.Event
 	switch {
 	case curObjPasses && !oldObjPasses:
-		return &watch.Event{Type: watch.Added, Object: event.Object.DeepCopyObject()}
+		watchEvent.Type = watch.Added
+		if _, ok := event.Object.(*CachingObject); ok {
```

Comment:
Having to do this test in many places seems very awkward.

Reply:
Sure - this code is POC quality - I was consciously making the simplest and ugly code.